waffle_utils/file/search.py

```diff
@@ -1,99 +1,73 @@
 from pathlib import Path
-from typing import List, Optional, Union
+from typing import Union
 
 from natsort import natsorted
 
-from waffle_utils.image import SUPPORTED_IMAGE_EXTENSIONS
-from waffle_utils.video import SUPPORTED_VIDEO_EXTENSION
+from waffle_utils.file.types import (
+    SUPPORTED_IMAGE_EXTENSIONS,
+    SUPPORTED_VIDEO_EXTENSION,
+)
 
 
 def get_files(
-    directory: Union[str, Path], extension: Union[list[str], str, None] = None
+    directory: Union[str, Path],
+    recursive: bool = True,
+    extension: Union[list[str], str, None] = None,
 ) -> list:
     """
     Retrieves a list of files in a directory, optionally filtered by extension.
 
     Args:
         directory (Union[str, Path]): Path to the directory.
-        extension (Union[list[str], str, None], optional): File extension to filter the files by. Defaults to None.
+        recursive (bool, optional): Whether to search recursively or not. Defaults to True.
+        extension (Union[list[str], str, None], optional): File extension(including ".") to filter the files by. Defaults to None.
 
     Returns:
         list: List of file paths.
     """
     directory = Path(directory)
 
-    if isinstance(extension, str):
-        extension = [extension]
-    elif extension is None:
-        extension = []
+    files = directory.glob(f"**/*" if recursive else "*")
+    files = list(filter(lambda file: file.is_file(), files))
+    if extension:
+        if isinstance(extension, str):
+            extension = [extension]
+        files = [file for file in files if file.suffix.lower() in extension]
```

Comment:
When we receive an extension, there is no guarantee that we will receive it in lowercase.

I think we should change the extension to lowercase (`lower()`) within the function.

Reply:
ok

Reply:
good